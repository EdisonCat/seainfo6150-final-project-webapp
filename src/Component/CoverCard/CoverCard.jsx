import React from "react";
import styles from "./CoverCard.module.css";
import { useHistory } from "react-router-dom";
export default function CoverCard(props) {
    const history = useHistory();
    const handleTrailer = () => {
        history.push({pathname: "/trailer", state: {url: props.show.url}});
    }
    return (
        <div className={styles.container} onClick={() => handleTrailer()}>
            <img className={styles.thumbnail} src={props.show.thumbnail} alt="thumbnail"/>
            <div className={styles.title}>
                {props.show.title}
            </div>
        </div>
    );
} 