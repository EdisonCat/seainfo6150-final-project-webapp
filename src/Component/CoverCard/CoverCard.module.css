.container {
    display: flex;
    flex-direction: column;
    align-items: center;
    cursor: pointer;
    margin: 10px 5px 10px 5px;
    border: 1px solid #333;
    transition: 1s;
}
.container:hover {
    border: 1px solid red;
}
.thumbnail {
    width: 200px;
    height: auto;
}
.title {
    text-align: center;
    margin-top: 10px;
}