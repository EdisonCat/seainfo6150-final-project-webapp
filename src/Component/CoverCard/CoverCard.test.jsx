import React from "react";
import CoverCard from "./CoverCard.jsx";
it("rendered CoverCard correctly", () => {
    const show = {
        thumbnail: "",
    };
	// eslint-disable-next-line no-undef
    const { container } = render(<CoverCard show={show}/>);
    expect(container).toMatchSnapshot();
});