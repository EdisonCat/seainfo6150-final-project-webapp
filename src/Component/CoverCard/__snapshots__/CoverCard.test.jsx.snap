// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered CoverCard correctly 1`] = `
<div>
  <div
    class="container"
  >
    <img
      alt="thumbnail"
      class="thumbnail"
      src=""
    />
    <div
      class="title"
    />
  </div>
</div>
`;
